### PR TITLE
tidb config: add txn-entry-size-limit

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What is changed, added or deleted? (Required)

Add a new configuration. It's introduced in v5.0 and v4.0.10. https://github.com/pingcap/tidb/pull/21843/files

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by cfzjywxk** [comment]
LGTM

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -303,6 +303,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 当 TiDB 检测到 tidb-server 的内存使用超过了阈值，则会认为存在内存溢出的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句以及 heap profile 记录到目录 [`tmp-storage-path/record`](/tidb-configuration-file.md#tmp-storage-path) 中，并输出一条包含关键字 `tidb-server has the risk of OOM` 的日志。
 + 该值作为系统变量 [`tidb_memory_usage_alarm_ratio`](/system-variables.md#tidb_memory_usage_alarm_ratio) 的初始值。
 
+### `txn-entry-size-limit` <span class="version-mark">从 v4.0.10 版本开始引入</span>
+
++ TiDB 单行数据大小限制
```
Comment (TomShawn) [inline tidb-configuration-file.md:308]:
```suggestion
+ TiDB 单行数据的大小限制
```

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -303,6 +303,12 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 默认值：6291456 (Byte)
++ 事务中单个 key-value 记录的大小限制，若超出该限制，TiDB 将会返回 `entry too large` 错误。该配置项的最大值不超过 `125829120`（表示 120MB）。注意，TiKV 同样有类似的限制，若单个写入请求的数据量大小超出 [`raft-entry-max-size`](/tikv-configuration-file.md#raft-entry-max-size)，默认为 8MB，TiKV 会拒绝处理该请求。当表的一行记录较大时，需要同时修改这 2 个配置。
```
Comment (TomShawn) [inline tidb-configuration-file.md:310]:
```suggestion
+ 事务中单个 key-value 记录的大小限制。若超出该限制，TiDB 将会返回 `entry too large` 错误。该配置项的最大值不超过 `125829120`（表示 120MB）。
+ 注意，TiKV 有类似的限制。若单个写入请求的数据量大小超出 [`raft-entry-max-size`](/tikv-configuration-file.md#raft-entry-max-size)，默认为 8MB，TiKV 会拒绝处理该请求。当表的一行记录较大时，需要同时修改这两个配置。
```

**Comment by TomShawn** [comment]
@youjiali1995 Looks like https://github.com/pingcap/tidb/pull/21843 will only be merged into release-4.0. Why should this PR be merged into master?

**Comment by youjiali1995** [comment]
> @youjiali1995 Looks like [pingcap/tidb#21843](https://github.com/pingcap/tidb/pull/21843) will only be merged into release-4.0. Why should this PR be merged into master?

It has merged into master. https://github.com/pingcap/tidb/pull/18231

**Review by TomShawn — APPROVED** [review]
LGTM

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -303,6 +303,13 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 当 TiDB 检测到 tidb-server 的内存使用超过了阈值，则会认为存在内存溢出的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句以及 heap profile 记录到目录 [`tmp-storage-path/record`](/tidb-configuration-file.md#tmp-storage-path) 中，并输出一条包含关键字 `tidb-server has the risk of OOM` 的日志。
 + 该值作为系统变量 [`tidb_memory_usage_alarm_ratio`](/system-variables.md#tidb_memory_usage_alarm_ratio) 的初始值。
 
+### `txn-entry-size-limit` <span class="version-mark">从 v4.0.10 版本开始引入</span>
```
Comment (TomShawn) [inline tidb-configuration-file.md:306]:
```suggestion
### `txn-entry-size-limit` <span class="version-mark">从 v5.0.0-rc 版本开始引入</span>
```

**Comment by TomShawn** [comment]
Follow-up: cherry-pick to v4.0 after v4.0.10 is released.

**Pull request merged**